### DISCUSSION TITLE
Advice to @woncheol-kim to upgrade from a Synology NAS

### DISCUSSION THREAD

**Discussion by qdm12** [qdm12/dns#80] [General]
@woncheol-kim From #79 

> Bridge network does not work on my old system. Interactive terminal for docker containers does not work either. It's kernel limitations, according to the link below:
> https://stackoverflow.com/questions/52520008/can-i-install-docker-on-arm8-based-synology-nas
> Now I'm really thinking of installing linux on my old i386 laptop and replace the arm32 NAS.

I had a Synology NAS with a x86 chip as well, but the system (old kernel, no package manager, old Docker, etc.) made me want to shoot myself as well back 3-4 years ago. Their OS upgrades lag way too behind. And you can't do anything really if your Kernel is too old and you have no package manager either sadly 😢 

What you could do (as I did more or less) is have the laptop as the 'thinking' piece and have it store data to your NAS over NFS (either use https://hub.docker.com/r/erichough/nfs-server/ or the Synology built-in NFS UI). You can put a small SSD in your laptop with your programs and docker-compose.yml files for example, and store all the bulk of you data on your NAS over the network. Performance should be almost identical for most uses.

I recommend you Alpine Linux for your laptop, or you can go the hardcore way with Arch Linux. In my opinion today, Debian/Ubuntu suck as they have old packages, plenty of security vulnerabilities etc.

**Comment by woncheol-kim** [comment]
Thank you for your advice. I've just ordered a USB ethernet adapter for my laptop. I was planning to install OpenSUSE, but just learned that Alpine Linux supports i386 and I'll try it. Debian/Ubuntu is out of my consideration because it does not support i386 any more.

**Reply by qdm12** [reply]
Cool! Definitely try Alpine, you should be able to just `setup-alpine` and then `apk add docker` to get it running, it's rather easy and fast to get setup in my opinion 😉 Let me know if you have any question, happy to help you out!

**Reply by woncheol-kim** [reply]
Hi, I've just installed alpine linux and have problem: `apk add docker` does not work. I've tried `apk add mc` and it works fine.

```
apk update
apk add docker
ERROR: unable to select packages:
  docker (no such package):
    required by: world[docker]

apk search docker
rsyslog-imdocker-8.2012.0-r0

cat /etc/apk/repositories
#/media/sdb/apks
http://dl-cdn.alpinelinux.org/alpine/v3.14/main
#http://dl-cdn.alpinelinux.org/alpine/v3.14/community
#http://dl-cdn.alpinelinux.org/alpine/edge/main
#http://dl-cdn.alpinelinux.org/alpine/edge/community
#http://dl-cdn.alpinelinux.org/alpine/edge/testing
```

Now what can I do?

**Reply by woncheol-kim** [reply]
Update: enabling the community repository solved the problem. :)
